network/p2p/network.go

```diff
@@ -516,7 +517,10 @@ func (n *Network) queueSubmitFunc(message interface{}) {
 
 	eng, err := n.subscriptionManager.GetEngine(qm.Target)
 	if err != nil {
-		logger.Err(err).Msg("failed to submit message")
+		// This means the message was received on a channel that the node has not registered an engine for
```

Comment:
but will the node ever receive a message on a channel it didn't subscribe to? wouldn't libp2p just discard that message without submitting it to the application?

Reply:
it's possible for unicast cases. There should be validation to prevent this, so it would be a bug/unexpected